### PR TITLE
feat(config): implement more config variables and add config verification

### PR DESCRIPTION
**Pull request by Baconing**
Add config variables, config verification, defaults, and some changes to docs.

Also fix a small bug with softdelete in the order model.

#19

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,151 @@
...
+| `database.password`     | `string`                      | The database password                                                                                              | No, unless `database.URL` is set, then it must be empty.  |
+| `database.database`     | `string`                      | The database name                                                                                                  | No, unless `database.URL` is set, then it must be empty.  |
+| `database.extraOptions` | `map` of `string` -> `string` | Extra options to pass to the database connection                                                                   | Yes, unless `database.URL` is set, then it must be empty. |
+| `database.URL`          | `string`                      | The database URL/DSN. A reference can be found [here](https://github.com/go-sql-driver/mysql#dsn-data-source-name) | Yes, unless other database entries are not set.           |
+| `tokensPerOrder`        | `int`                         | The amount of tokens a user gets per order. Must be > 0 and defaults to 1.                                         | Yes                                                       |
```
Comment (refractored) [inline README.md]: user pays per order*

File: config.json.example [hunk 2]
```diff
@@ -1,6 +1,13 @@
 {
-  "token": "BOT_TOKEN",
-  "mysqldsn": "user:password@tcp(localhost:3306)/your_database_name?parseTime=true",
-  "owners": ["1234", "1234"],
-  "guildid": "1234"
+    "token": "secret",
+    "owners": ["you", "and", "your", "friends"],
+    "guildID": "some numbers",
+
```
Comment (refractored) [inline config.json.example:5]: the StartupChannelID config option is not present

File: README.md [hunk 3]
```diff
@@ -1,26 +1,151 @@
...
+|-------------------------|-------------------------------|--------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------|
+| `token`                 | `string`                      | The Discord bot token                                                                                              | No                                                        |
+| `owners`                | `array` of `string`           | The Discord IDs of the bot owners                                                                                  | One is required                                           |
+| `guildID`               | `string`                      | The Discord ID of the guild that certain commands will be registered in (e.g. `/blacklist`)                        | No                                                        |
+| `database`              | `object`                      | The database configuration                                                                                         | No                                                        |
```
Comment (refractored) [inline README.md:60]: StartupChannelID is also missing here aswell

**Review by refractored — CHANGES_REQUESTED** [review]

File: config.json.example (hunk 2 above)
Comment (Baconing) [reply config.json.example:5]: intended

**Review by refractored — APPROVED** [review]

**Pull request merged**